### ISSUE TITLE
cdf for MixtureModel{Univariate}

### ISSUE THREAD

**Issue by JockLawrie** [JuliaStats/Distributions.jl#369]
Hi there,

I have written a cdf for univariate mixture models defined on the real line - see below.
The underlying math is valid for any other univariate mixture model, however I can't get the function working with x being a more generic type than Float64. For example, x::Int should work for mixtures of Poisson distributions. How can I get it working for say x::Real or x::Number, or just x?

Cheers,
Jock

```julia
import Distributions.cdf    # Necessary to extend cdf to MixtureModel types

function cdf(d::MixtureModel{Univariate}, x::Float64)
    # cdf for UnivariateMixture
    result = 0.0
    comps  = components(d)
    pr     = probs(d)
    nc     = size(comps, 1)    # Number of componenets
    for i = 1:nc
        result += (pr[i] * cdf(comps[i],  x))
    end
    result
end
```

**Comment by simonbyrne** [comment]
The problem is that if you define a `cdf(d::MixtureModel{Univariate}, x::Real)` method, you get ambiguous dispatch warnings. You can try defining
```julia
cdf(d::MixtureModel{Univariate,Discrete},x::Int) = invoke(cdf, (MixtureModel{Univariate},Real), d, x)
```
first, then define
```julia
function cdf(d::MixtureModel{Univariate}, x::Real)
    # cdf for UnivariateMixture
    result = 0.0
    comps  = components(d)
    pr     = probs(d)
    nc     = size(comps, 1)    # Number of componenets
    for i = 1:nc
        result += (pr[i] * cdf(comps[i],  x))
    end
    result
end
```

**Comment by JockLawrie** [comment]
Great, that works, thanks Simon.
And apologies for posting here, this is a julia-users question rather than an issue with the Distributions package.

**Comment by lindahua** [comment]
Using the new interface, now you should use ``ncomponents(d)`` to get the number of components, and `component(d, i)`` to get the ``i``-th component. This allows you to work with ``AbstractMixtureModel`` instead of just the specific ``MixtureModel`` type.

I will add the ``cdf`` soon, and update the document.

**Comment by lindahua** [comment]
Implemented by f35ac5f
